core/rsyslog/config_test.go

```diff
@@ -0,0 +1,151 @@
+
+	"github.com/juju/juju/core/rsyslog"
+)
+
+func TestPackage(t *testing.T) {
+	gc.TestingT(t)
+}
+
+var _ = gc.Suite(&configSuite{})
+
+const (
+	cert = `
+-----BEGIN CERTIFICATE-----
+MIIBjDCCATigAwIBAgIBADALBgkqhkiG9w0BAQUwHjENMAsGA1UEChMEanVqdTEN
+MAsGA1UEAxMEcm9vdDAeFw0xMjExMDkxNjQwMjhaFw0yMjExMDkxNjQ1MjhaMB4x
+DTALBgNVBAoTBGp1anUxDTALBgNVBAMTBHJvb3QwWTALBgkqhkiG9w0BAQEDSgAw
+RwJAduA1Gnb2VJLxNGfG4St0Qy48Y3q5Z5HheGtTGmti/FjlvQvScCFGCnJG7fKA
+Knd7ia3vWg7lxYkIvMPVP88LAQIDAQABo2YwZDAOBgNVHQ8BAf8EBAMCAKQwEgYD
+VR0TAQH/BAgwBgEB/wIBATAdBgNVHQ4EFgQUlvKX8vwp0o+VdhdhoA9O6KlOm00w
+HwYDVR0jBBgwFoAUlvKX8vwp0o+VdhdhoA9O6KlOm00wCwYJKoZIhvcNAQEFA0EA
+LlNpevtFr8gngjAFFAO/FXc7KiZcCrA5rBfb/rEy297lIqmKt5++aVbLEPyxCIFC
+r71Sj63TUTFWtRZAxvn9qQ==
+-----END CERTIFICATE-----
+`
+
+	key = `
+-----BEGIN RSA PRIVATE KEY-----
+MIIBOQIBAAJAduA1Gnb2VJLxNGfG4St0Qy48Y3q5Z5HheGtTGmti/FjlvQvScCFG
+CnJG7fKAKnd7ia3vWg7lxYkIvMPVP88LAQIDAQABAkEAsFOdMSYn+AcF1M/iBfjo
+uQWJ+Zz+CgwuvumjGNsUtmwxjA+hh0fCn0Ah2nAt4Ma81vKOKOdQ8W6bapvsVDH0
+6QIhAJOkLmEKm4H5POQV7qunRbRsLbft/n/SHlOBz165WFvPAiEAzh9fMf70std1
+sVCHJRQWKK+vw3oaEvPKvkPiV5ui0C8CIGNsvybuo8ald5IKCw5huRlFeIxSo36k
+m3OVCXc6zfwVAiBnTUe7WcivPNZqOC6TAZ8dYvdWo4Ifz3jjpEfymjid1wIgBIJv
+ERPyv2NQqIFQZIyzUP7LVRIWfpFFOo9/Ww/7s5Y=
+-----END RSA PRIVATE KEY-----
+`
+
+	invalidKey = `
+-----BEGIN RSA PRIVATE KEY-----
+MIIBOQIBAAJBAJkSWRrr81y8pY4dbNgt+8miSKg4z6glp2KO2NnxxAhyyNtQHKvC
++fJALJj+C2NhuvOv9xImxOl3Hg8fFPCXCtcCAwEAAQJATQNzO11NQvJS5U6eraFt
+FgSFQ8XZjILtVWQDbJv8AjdbEgKMHEy33icsAKIUAx8jL9kjq6K9kTdAKXZi9grF
+UQIhAPD7jccIDUVm785E5eR9eisq0+xpgUIa24Jkn8cAlst5AiEAopxVFl1auer3
+GP2In3pjdL4ydzU/gcRcYisoJqwHpM8CIHtqmaXBPeq5WT9ukb5/dL3+5SJCtmxA
+jQMuvZWRe6khAiBvMztYtPSDKXRbCZ4xeQ+kWSDHtok8Y5zNoTeu4nvDrwIgb3Al
+fikzPveC5g6S6OvEQmyDz59tYBubm2XHgvxqww0=
+-----END RSA PRIVATE KEY-----
+`
+
+	invalidCert = `
+-----BEGIN CERTIFICATE-----
+MIIBOgIBAAJAZabKgKInuOxj5vDWLwHHQtK3/45KB+32D15w94Nt83BmuGxo90lw
+-----END CERTIFICATE-----
+`
+)
+
+type configSuite struct{}
+
+func (s *configSuite) TestValidation(c *gc.C) {
+	tests := []struct {
+		about  string
+		url    string
+		caCert string
+		cert   string
+		key    string
+		err    string
+	}{{
+		about: "fail to parse url",
+		url:   "%",
+		err:   "cannot parse rsyslog url: parse %: invalid URL escape \"%\"",
+	}, {
+		about: "invalid url schema",
+		url:   "ftp://test.com",
+		err:   "url schema is not https: ftp not valid",
+	}, {
+		about:  "invalid ca cert",
+		caCert: invalidCert,
+		err:    "cannot parse rsyslog CA cert: asn1: syntax error: data truncated",
+	}, {
+		about: "invalid cert",
+		cert:  invalidCert,
+		err:   "cannot parse rsyslog client cert: asn1: syntax error: data truncated",
+	}, {
+		about: "invalid key/cert pair",
+		key:   invalidKey,
+		err:   "invalid rsyslog client cert/key pair: crypto/tls: private key does not match public key",
+	}, {
+		about: "ok",
+	},
+	}
+	for i, test := range tests {
+		c.Logf("running test %d: %v", i, test.about)
+		config := rsyslog.ClientConfig{
```

Comment:
ymmv, but search for checkNotValid, I've found the

    func (*Suite) TestURLSchema(c *gc.C) {
        config := minimalValidConfig()
        config.URL = "http://www.google.com/"
        checkNotValid(c, config, "URL not valid: https required")
    }

approach rather more tractable than table tests in the long term